File: python/2.py
# Definition for singly-linked list.
class ListNode(object):
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution(object):
    def addTwoNumbers(self, l1, l2):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """
        i = 0
        a = 0
        while l1 != None :
            i = i + l1.val * 10 ** a
            l1 = l1.next
            a += 1
        j = 0
        a = 0
        while l2 != None :
            j = j + l2.val * 10 ** a
            l2 = l2.next
            a += 1
        i += j
        print i
        head = ListNode(i % 10)
        now = head
        i /= 10
        while i != 0 :
            now.next = ListNode(i % 10)
            now = now.next
            i /= 10
        return head


a1 = ListNode(1)
a2 = ListNode(2)
a3 = ListNode(3)
a1.next = a2
a2.next = a3
b1 = ListNode(1)
b2 = ListNode(1)
b3 = ListNode(1)
b1.next = b2
b2.next = b3
s = Solution()
s.addTwoNumbers(a1,b1)

File: cpp/2.cpp
//2. Add Two Numbers
//        You are given two non-empty linked lists representing two non-negative integers. The digits are stored in reverse order and each of their nodes contain a single digit. Add the two numbers and return it as a linked list.
//
//You may assume the two numbers do not contain any leading zero, except the number 0 itself.
//
//Input: (2 -> 4 -> 3) + (5 -> 6 -> 4)
//Output: 7 -> 0 -> 8

#include <iostream>
#include <vector>
#include <map>

using namespace std;

// * Definition for singly-linked list.
struct ListNode {
    int val;
    ListNode *next;
    ListNode(int x) : val(x), next(NULL) {}
};

class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        int c = 0,s = 0;
        struct ListNode head(0);
        struct ListNode *current = &head;
        while (l1 || l2 || c){
            s = c;
            if (l1) {
                s += l1->val;
                l1 = l1->next;
            }
            if (l2) {
                s += l2->val;
                l2 = l2->next;
            }
            c = s / 10;
            s %= 10;
            current->next = new struct ListNode(s);
            current = current->next;
        }
        return head.next;
    }
};

int main() {
    std::cout << "Hello, World!" << std::endl;
    return 0;
}

